src/test/java/de/tum/in/www1/artemis/ArchitectureTest.java

```diff
@@ -160,14 +183,22 @@ public void check(JavaMethod item, ConditionEvents events) {
         };
     }
 
-    /**
-     * Checks that no class directly calls Git.commit(), but instead uses GitService.commit()
-     * This is necessary to ensure that committing is identical for all setups, with and without commit signing
-     */
-    @Test
-    void testNoDirectGitCommitCalls() {
-        ArchRule usage = noClasses().should().callMethod(Git.class, "commit").because("You should use GitService.commit() instead");
-        var classesWithoutGitService = allClasses.that(not(assignableTo(GitService.class)));
-        usage.check(classesWithoutGitService);
+    private ArchCondition<JavaMethod> haveAllParametersAnnotatedWithUnless(DescribedPredicate<? super JavaAnnotation<?>> annotationPredicate,
+            DescribedPredicate<JavaClass> exception) {
+        return new ArchCondition<>("have all parameters annotated with " + annotationPredicate.getDescription()) {
+
+            @Override
+            public void check(JavaMethod item, ConditionEvents events) {
+                boolean satisfied = item.getParameters().stream()
+                        // Ignore annotations of the Pageable parameter
+                        .filter(javaParameter -> !exception.test(javaParameter.getRawType())).map(JavaParameter::getAnnotations)
```

Comment:
I don't quite understand how getRawType() (only) filters out the pageable parameters, can you elaborate a bit on that?